### PR TITLE
TABU list algorithm with BIC score

### PR DESCRIPTION
**Pull request by MLasserre**
Adds a new score based learning algorithm for CBNs.
Includes documentation and tests.

### CODE REVIEW

File: lib/src/otagrum/TabuList.hxx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+#include "otagrum/IndicesManip.hxx"
+
+namespace OTAGRUM
+{
+
+class OTAGRUM_API TabuList : public OT::Object
+{
+public:
+  // Constructor
+  explicit TabuList(const OT::Sample &data, const NamedDAG &initial_dag,
```
Comment (jschueller) [inline lib/src/otagrum/TabuList.hxx:21]: no need for explicit here as there a 2 non optional parameters

File: lib/src/otagrum/TabuList.hxx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                    const unsigned int max_parents = 4, const unsigned int restarts = 1,
+                    const unsigned int tabu_list_size = 2);
+  explicit TabuList(const OT::Sample &data,
+                    const unsigned int max_parents = 4,
+                    const unsigned int restarts = 1,
+                    const unsigned int tabu_list_size = 2);
```
Comment (jschueller) [inline lib/src/otagrum/TabuList.hxx:27]: you could use OT::UnsignedInteger instead of int (always 32 bits)

Comment (MLasserre) [reply lib/src/otagrum/TabuList.hxx:27]: It's done.
Do you want me to do the same for the remainder of the file ?
Like for the attributes for example.

Comment (jschueller) [reply lib/src/otagrum/TabuList.hxx:27]: please

**Review by jschueller — APPROVED** [review]

**Pull request merged**